### DISCUSSION TITLE
Obtain License types of dependencies for a given spec

### DISCUSSION THREAD

**Discussion by a-jp** [spack/spack#46300] [Q&A]
I've used spack to build my toolchain (codes I need to now build my own codebase) and in doing so spack will clearly (and has) pulled in dependencies as it's designed to do. I've been asked to audit and show that my code base that will use this spack produced toolchain does not have specific open-source license types.  Is there away to ask spack to walk the dependency graph and output/print the license types that are used across the toolchain I've just built?  I can then use this to audit that I've not used a third party library that has a specific open-source license type?  I'm thinking spack may be able to do something like this as it understands the dependency graph very well.  All help welcome.  I'm thinking something along the lines of pip-licenses...
Thanks

**Comment by a-jp** [comment]
Hi all, I wondered if anyone could point me in the right direction with this?  Are there any other bits of information I could usefully add that would help clarify my question? Many thanks

**Comment by tgamblin** [comment]
There's not currently a way to get all the license information post-concretization.  We *do* have license information, but we don't currently store it as an attribute on concrete specs.  Really it should be added to the info in #32312, and should likely be included in the package hash.  Once we add that, we could easily show it in `spack spec` and in `spack spec --json`

**Reply by a-jp** [reply]
Hi, doing so via say 'spack spec -json' would be enormously useful and would allow one to confirm licence types trivially.  I do not have the skills to help, but I'm happy to help test.  Thanks

**Reply by tgamblin** [reply]
I can do this as a follow-on to #32312, hopefully soon-ish.  Note that one way you could help is to help confirm license entries in spack `package.py` files.  You can add a `license(..., checked_by="github_user")` `kwarg` to the package to indicate that you checked the license. We'd like to have all of them checked eventually -- they were added via an automatic process.
